python/test/unit/language/test_decorator.py

```diff
@@ -0,0 +1,24 @@
+import triton
+import pytest
+
+def test_decorator_with_def(device):
+    def triton_heuristics_pointwise(**kwargs):
+        def decorator(func):
+            return func
+
+        return decorator
+
+    @triton_heuristics_pointwise(
+        size_hints=[64],
+        filename=__file__,
+        inductor_meta={'backend_hash': '8cde1805b523918efde951fadd02ae7fb36906d8f746f2adef0aeffabe53b3f8'},
+        min_elem_per_thread=0
+    )
```

Comment:
can we minimize this testcase a little and add a brief comment explaining what's being tested?  In particular this is nonobvious because the "def" in question seems to be inside the random hash.